Library/MapleBacon/MapleBacon/Resizer.swift

```diff
@@ -6,176 +6,251 @@ import UIKit
+    private struct ResizeCalculationEntity {
+        
+        var currentSize: CGSize  = CGSize()
+        var intendedSize: CGSize = CGSize()
+        
+        var horizontalRatio: CGFloat = 0
+        var verticalRatio: CGFloat = 0
+        var newSize: CGSize = CGSize()
+        var offset: CGSize = CGSize()
+        var offsetPoint: CGPoint = CGPoint()
+        
+        var resizeBounds: CGRect {
+            return CGRect(x: self.offsetPoint.x, y: self.offsetPoint.y, width: self.intendedSize.width, height: self.intendedSize.height)
+        }
+        
+        init(currentSize imgSize: CGSize, intendedSize size: CGSize) {
+            
+            self.currentSize = imgSize
+            self.intendedSize = size
+            
+            self.horizontalRatio = size.width / imgSize.width
+            self.verticalRatio = size.height / imgSize.height
+            self.offset = CGSize(width: imgSize.width - size.width, height: imgSize.height - size.height)
+            self.offsetPoint = CGPoint(x: self.offset.width / 2.0, y: self.offset.height / 2.0)
+        }
+        
+        mutating func applyContentMode(contentMode: UIViewContentMode) {
+            switch contentMode {
+            case .scaleToFill, .redraw:
+                self.newSize = CGSize(width: self.currentSize.width * self.horizontalRatio, height: self.currentSize.height * self.verticalRatio)
+                self.offset  = self.calcOffset(fromSize: self.newSize, toSize: self.intendedSize)
+            case .scaleAspectFill:
+                let ratio = max(self.horizontalRatio, self.verticalRatio)
+                newSize = CGSize(width: self.currentSize.width * ratio, height: currentSize.height * ratio)
+                offset = self.calcOffset(fromSize: newSize, toSize: self.intendedSize)
+            case .scaleAspectFit:
+                let ratio = min(self.horizontalRatio, self.verticalRatio)
+                newSize = CGSize(width: self.currentSize.width * ratio, height: self.currentSize.height * ratio)
+                offset = self.calcOffset(fromSize: newSize, toSize: self.intendedSize)
+            case .center:
+                break
+            case .top:
+                self.offsetPoint.y = 0
+            case .bottom:
+                self.offsetPoint.y = self.offset.height
+            case .left:
+                self.offsetPoint.x = 0
+            case .right:
+                self.offsetPoint.x = self.offset.width
+            case .topLeft:
+                self.offsetPoint = CGPoint(x: 0, y: 0)
+            case .topRight:
+                self.offsetPoint = CGPoint(x: self.offset.width, y: 0)
+            case .bottomLeft:
+                self.offsetPoint = CGPoint(x: 0, y: self.offset.height)
+            case .bottomRight:
+                self.offsetPoint = CGPoint(x: self.offset.width, y: self.offset.height)
+            }
+        }
+        
+        private mutating func calcOffset(fromSize size: CGSize, toSize: CGSize) -> CGSize {
```

Comment:
also why is a function that does a calculation and returns something mutating the struct? Side effect is unexpected. This could return a tuple